libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/NetworkMonitor.kt

```diff
@@ -0,0 +1,50 @@
+package com.apollographql.apollo3.network
+
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.takeWhile
+import okio.Closeable
+import kotlin.js.JsName
+
+internal interface PlatformNetworkMonitor: Closeable {
+  fun setListener(listener: Listener)
+
+  interface Listener {
+    fun networkChanged(isOnline: Boolean)
+  }
+}
+
+internal expect fun platformNetworkMonitor(): PlatformNetworkMonitor?
+
+@ApolloExperimental
+interface NetworkMonitor: Closeable {
```

Comment:
Good call. I renamed `PlatformNetworkMonitor` to `ConnectivityManager` in https://github.com/apollographql/apollo-kotlin/pull/5690/commits/92ad158c1d9c527d3c2db275c082bc00a14537ba.

The important one is `NetworkMonitor` because it's the public one. I'm not sold on the name so we could swap or use anything else but I think I prefer the "short" name and not prefix everything with `Apollo`, only those symbol with a high risk of clash (`Request`, `Call`, `Client`, `Interceptor`). I think `NetworkMonitor` is fine? But no strong opinion there.

Reply:
`PlatformConnectivityManager` is 👍 . All good!